hetio/neo4j.py

```diff
@@ -5,21 +5,31 @@
 from operator import or_
 from functools import reduce
 
-import py2neo
-import py2neo.packages.httpstream
 import pandas
 from tqdm import tqdm
 
 import hetio.hetnet
 
-# Get py2neo version
-PY2NEO_VER = int(py2neo.__version__[0])
 
-# Avoid SocketError
-py2neo.packages.httpstream.http.socket_timeout = 1e8
+@functools.lru_cache()
+def import_py2neo():
+    """
+    Imports the py2neo library, checks its version, and sets the socket timeout if necessary
+    """
+    import py2neo
+    # Get py2neo version
+    PY2NEO_VER = int(py2neo.__version__[0])
```

Comment:
should we do `PY2NEO_VER = tuple(int(x) for x in py2neo.__version__.split('.'))`. Then `if PY2NEO_VER < (4, 0)`. This is more resilient if major version number becomes double digits.

Reply:
Actually it's probably best to just use 

```python
import pkg_resources
version = pkg_resources.parse_version(py2neo.__version__)
if version._version.release[0] < 4:
    # blah
```


https://setuptools.readthedocs.io/en/latest/pkg_resources.html#parsing-utilities

Reply:
And perhaps import_py2neo should return `py2neo, version_tuple` where version_tuple is `version._version.release`. Then this can be used in other functions.